### PR TITLE
Modularize the lib, just throwing an idea

### PR DESCRIPTION
**Pull request by portellaa**
* Restructure filesystem
* Update Package.swift to support modules

#6

### CODE REVIEW

**Comment by portellaa** [comment]
Still have to update the README file

File: PubSub/Tests/LinuxMain.swift [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import XCTest
```
Comment (Andrewangeta) [inline PubSub/Tests/LinuxMain.swift]: We can remove the PubSub folders since it's not implemented and we can move these files to a common location.

**Review by Andrewangeta — CHANGES_REQUESTED** [review]

File: PubSub/Tests/LinuxMain.swift (hunk 1 above)
Comment (Andrewangeta) [reply PubSub/Tests/LinuxMain.swift]: Other than that it looks good to me.

Comment (portellaa) [reply PubSub/Tests/LinuxMain.swift]: Sorry, i forgot those 🙇 will change in the mean while

**Comment by portellaa** [comment]
@Andrewangeta updated 😉

**Comment by Andrewangeta** [comment]
LGTM

**Review by Andrewangeta — APPROVED** [review]

**Pull request merged**